source/client/options_impl.cc

```diff
@@ -371,13 +371,12 @@ CommandLineOptionsPtr OptionsImpl::toCommandLineOptions() const {
     // TODO(oschaaf): expose append option in CLI? For now we just set.
     header_value_option->mutable_append()->set_value(false);
     auto request_header = header_value_option->mutable_header();
-    std::vector<std::string> split_header = absl::StrSplit(
-        header, ':',
-        absl::SkipWhitespace()); // TODO(oschaaf): maybe throw when we find > 2 elements.
-    request_header->set_key(split_header[0]);
-    if (split_header.size() == 2) {
-      request_header->set_value(split_header[1]);
-    }
+    auto pos = header.find(':');
+    if (pos != std::string::npos) {
+      request_header->set_key(header.substr(0, pos));
+      // Any visible char, including ':', is allowed in header values.
+      request_header->set_value(header.substr(pos + 1));
+    } // else throw? a ':' is required in a header.
```

Comment:
+1, e.g. `throw MalformedArgvException("A ':' is required in a header");` ?
It would be great to also nail the case down in a test in `options_test.cc`. This would also help with keeping coverage up. Some pseudo-code for that:

```
TEST_F(OptionsImplTest, RequestHeaderWithoutColon) {
  EXPECT_THROW_WITH_REGEX(TestUtility::createOptionsImpl("client --request-header bar",
                          MalformedArgvException, "is required in a header");
}
```

Reply:
Thanks for the review, I've added the exception and test.